src/bin/julialauncher.rs

```diff
@@ -166,32 +166,49 @@ fn get_julia_path_from_channel(
     juliaupconfig_path: &Path,
     juliaup_channel_source: JuliaupChannelSource,
 ) -> Result<(PathBuf, Vec<String>)> {
+    let potential_matches: Vec<_> = config_data
+        .installed_channels
+        .keys()
+        .filter(|&item| {
+            item.starts_with(channel) && !item.starts_with("0") && !item.starts_with("1")
```

Comment:
This is more general:
```suggestion
            item.starts_with(channel) && !item.chars().next().map(char::is_numeric).unwrap_or(false)
```